frontend/utils.mjs

```diff
@@ -241,49 +240,48 @@ export async function buildOrWatch(config) {
         }
         buildsInProgress++
         onBuildStart?.(config)
-        reloadLiveServer()
         buildPromise = runBuild()
         const buildResponse = await buildPromise
         buildPromise = null
         await onBuildComplete?.(config, buildResponse)
         buildsInProgress--
         if (buildsInProgress === 0) {
             server?.resumeServer()
+            reloadLiveServer()
         }
+
         if (isDev && buildAgain) {
             void debouncedBuild()
         }
     }
 
-    let result = null
+    let esbuildContext = null
     let buildCount = 0
+    const activitySuccess = (count) => (count === 1 ? 'Built' : 'Rebuilt')
+    const activityError = (count) => (count === 1 ? 'Building failed' : 'Rebuilding failed ')
 
     async function runBuild() {
+        if (!esbuildContext) {
+            esbuildContext = await context({ ...commonConfig, ..._config })
+        }
```

Comment:
esbuild now doesn't run a build with config as previously

instead you create a context and call rebuild on it (never build)

---

you can also call serve and watch but not quite in such a way that we could replace express and chokidar